### ISSUE TITLE
The Gradle plugin is not compatible with Gradle 7.0

### ISSUE THREAD

**Issue by sergeykad** [joelittlejohn/jsonschema2pojo#1070]
The following message is printed if using the plugin with Gradle 6.0.1:
`Property 'configuration' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.`

https://docs.gradle.org/6.7/userguide/more_about_tasks.html#sec:up_to_date_checks

**Comment by joelittlejohn** [comment]
If you have a suggestion for fixing this, I'd be happy to accept a PR.

**Comment by sergeykad** [comment]
I assume that  `@Input` should be added to the `GenerationConfig configuration
` in the `GenerateJsonSchemaJavaTask`, but additional changes may be required.

**Comment by eirnym** [comment]
The plugin in this repository is compatible with gradle 2.3-5.6 without any warnings. You can check out https://plugins.gradle.org/plugin/com.github.eirnym.js2p for the gradle 5.6+.

**Comment by joelittlejohn** [comment]
@eirnym Would you be able to contribute back to this project and make our gradle plugin compatible with recent versions of gradle? Even just an explanation of the steps required would be useful.

**Comment by eirnym** [comment]
@joelittlejohn the https://github.com/eirnym/js2p-gradle/issues/2#issuecomment-639323829 lists all the reasons I don't try it without any discussion with you. Also I've tried to communicate with you via email a while ago on this matter as well without any luck. 

TD;LR 
* I support the latest version of the library 
* there's no easy and fast way to contribute back in the way the library is currently written
* we can use a model like checker framework, where they have a gradle plugin outside the repository
* Obviously, we can discuss the details if you want

**Comment by joelittlejohn** [comment]
Okay, this sounds like quite a serious blocker:

> First is quite obvious. it's support for official gradle repo plugins.gradle.org. It's possible to publish there from Gradle, but not from Maven, or at least I haven't found proper way to proceed with whole procedure using just Maven.
> Nevertheless, it can be easily done in the opposite way, converting all POMs to gradle.Looking on how the pom.xml files are organised in the original repository, I don't believe that the author will convert all POMs to build.gradle. Also I see that transformation as a long way as it doesn't guarantee the author have the same or even similar POMs in the maven repository. This is the second obstacle.

I think I might just refer to your plugin in the README.md of this project. My only goal is that people can easily find out about what they should do, which we might be able to achieve with just a change in the readme 🙂

**Comment by eirnym** [comment]
Sure, I can rename plugin as well if you agree with it

**Comment by AlexTrotsenko** [comment]
@eirnym , @joelittlejohn  thanks guys for having the discussion. I hope it will lead to the solution, which will be beneficial for the community in general and both of you in particular .

In order to speed-up the conversion  may I possibly ask you some questions/suggestions ?
1.  @joelittlejohn what do think about gradle as build system comparing to maven?
I have noticed that now more and more people prefer gradle as the build system instead of the maven.
E.g.  [Spring Boot has recently completed their migration to Gradle](https://spring.io/blog/2020/06/08/migrating-spring-boot-s-build-to-gradle).
2. @eirnym  have I understood correctly, that the main issue is the maven, which is currently used in [jsonschema2pojo-gradle-plugin](https://github.com/joelittlejohn/jsonschema2pojo/tree/master/jsonschema2pojo-gradle-plugin) module ? If it would be gradle - would it be possible to unblock the development?
3. @eirnym  if above mentioned point is the case, but @joelittlejohn  is not ready to migrate maven to gradle in the whole project - should it be enough then to have _only_  "gradle-plugin" sub-module migrateed to  gradle  build while keeping maven in the rest of the project ?
@joelittlejohn , what is your vision about it?
4. if none of the solutions mentioned above is possible or complicated for one of you - what do you think about creating new free organisation called `jsonschema2pojo` and moving @eirnym's  repo there and making it "official" gradle repo for `jsonschema2pojo`?

E.g. I see this approach is being used by AssertJ:
- core of assertJ is here https://github.com/joel-costigliola/assertj-core
- and here is an organisation with more related projects/repos: https://github.com/assertj

**Comment by eirnym** [comment]
@AlexTrotsenko 
As I mentioned migration to gardle is a huge stress which could lead to POM files placed in repository will be not the same as author and users want

And there's no obstacle to use an official plugin written and supported by other author in the separate repository. This model is well used in many popular libraries like Lombok, checker framework, and any xic plugin is a third-party. You mentioned you worry about freshness of plugin, while it's not an issue.

I don't understand why you want to force merge these two repos while it's not required

**Comment by joelittlejohn** [comment]
Hi @AlexTrotsenko, re your questions:

I haven't used Gradle much, although I can see that something has happened in the last couple of years and although Gradle has been around long time, suddenly adoption has accelerated. Maybe related to Android. I expect Maven is still more widely in Java FOSS projects but in 2020 it looks like the direction is towards Gradle. I don't really want to migrate this project to Gradle, I have to admit.

I like both your suggestions 3 and 4 because I think they are clear for new users, but if @eirnym would rather keep this code where it is, that's fine too. The main thing we need here is a clear recommendation/link for people who are looking for this, and it would also be nice if the gradle plugin was named in a way that people expect (and can naturally find). I'm happy to discontinue development of older versions of the 'legacy' Gradle plugin - people can continue to use the previously published versions if they need to.

**Comment by AlexTrotsenko** [comment]
@eirnym according to what I normally see, single repo is always easier for the end users to discover and navigate and also better for the authors of modules to keep them in sync.
Also this repo used to have a gradle plugin in the same repo - so it was natural suggestion to keep it there as well.

But keeping it as a separate repo is a good choice as well if the single repo approach will cause an inconvenience for you as a main contributor and given in to account the latest info mentioned by @joelittlejohn about migration to gradle. 

I would only suggest to think about discoverability of this new plugin repo. 

E.g. I could suggest to:
- make a bold notice in the current repo, that current gradle plugin is deprecated and should be used only for gradle < 5.6.
- add the link to the new gradle plugin, which should be used with gradle 5.6 +
- rename new plugin in gradle plugin portal to match the name of existing plugin in order to indicate, that it's official plugin. At least something with `jsonschema2pojo` in the name.
- additionally consider moving new gradle plugin to new free organisation called `jsonschema2pojo` _(as in my previous example with https://github.com/assertj )_

Does it make sense to you?

**Comment by joelittlejohn** [comment]
@AlexTrotsenko Yep, completely agree with your suggestions. I haven't had time today to update the README.md in this project, but I do plan to include a clear link there that tells people what to use for each version of Gradle.

It starts.... https://github.com/jsonschema2pojo 😄

**Comment by AlexTrotsenko** [comment]
@joelittlejohn  thanks for creating the organisation. 

Meanwhile, in order to facilitate the process, I have created the PR https://github.com/joelittlejohn/jsonschema2pojo/pull/1141, which _(as earlier discussed)_ deprecates gradle plugin in the current repo and adds the link to the new repo.
Please check, feel free to update _(if you find it necessary)_  and merge when you have a moment.

@eirnym could you please update [your repo](https://github.com/eirnym/js2p-gradle) so that it's written, that it's _official_ Gradle plugin of https://github.com/joelittlejohn/jsonschema2pojo.

Could you please update the repo and plugin name from `js2p` to `jsonschema2pojo` so that it's easier to discover and clear for everyone? 

As far as I know GitHub will automatically redirect from old name  https://github.com/eirnym/js2p-gradle to the new one.

Feel free to check what should be done on the Gradle plugin portal: e.g. adding notice and the link in https://plugins.gradle.org/plugin/com.github.eirnym.js2p or redirect if possible, etc.

Please keep us posted if you would like to keep your gradle repo in your github account for now or would you prefer to manage it in newly created [jsonschema2pojo](https://github.com/jsonschema2pojo) organisation.

**Comment by AlexTrotsenko** [comment]
@joelittlejohn thanks for merging the MR.

@eirnym  I have created the MR in `js2p-gradle`  repo as well:  https://github.com/eirnym/js2p-gradle/pull/3

I have moved earlier mentioned points from my previous comment to the issue in the same repo: https://github.com/eirnym/js2p-gradle/issues/4

**Comment by ben-manes** [comment]
Sorry I didn't know of this thread earlier. I only stumbled on it tonight.

I wrote the original gradle plugin for js2p / flyway / jOOQ, and then contributed them upstream. My intent wasn't a single repo, but that the parent projects would evolve with active stakeholder and shouldn't be dependent on an external developer who may disappear. I think @joelittlejohn should be the owner of whatever repository develops the plugin and he should have all of the credentials needed. You might consider promoting js2p to a github organization if you plan on a multi-repo approach.

You should be able to deploy to the portal from Maven. When the plugin portal first came about, one had to publish to Bintray and enable synchronization. This is how my [gradle-versions-plugin](https://github.com/ben-manes/gradle-versions-plugin) still works. The build has no concept of the portal and uses the bintray plugin, which is available for Maven. Flyway is also on maven and the portal, but I don't know how they set that up as it was much later. That said, it felt very hacky since the `gradleApi()` dependency needs to be available. Personally I'd use Github Actions to deploy with and let each plugin use their own build, handling that disparity in the CI.

It would be nice if the plugin maintained the same coordinates as it isn't discoverable for those who have it in our builds and expect our update checks to tell us of new versions. A major version change that breaks compatibility and changes the coordinate is perfectly okay.

I like that someone is taking over the plugin, it was a weekend project that could be much better. But given the ebb and flow of contributors in OSS, I strongly feel that @joelittlejohn should own the repository and have all the credentials to ensure a cohesive project.

**Comment by eirnym** [comment]
I'm sorry for a long break. I'm back and ready to fulfil my promises, if they're still actual :)

**Comment by eirnym** [comment]
The 'org.jsonschema2pojo' plugin version 1.1 has been published. Past year was tough, but we made it :)

**Comment by eirnym** [comment]
@joelittlejohn Do we need an old gradle plugin in this repo?

**Comment by joelittlejohn** [comment]
@eirnym Thank you for your recent contributions and help with commenting on issues.

I will be honest, I was not expecting you to publish to plugins.gradle.org using the org.jsonschema2pojo **package**. I was happy to indicate in this project that your plugin is preferred, but the way things are published right now is causing confusion and it makes me uncomfortable that the main identity of the project at plugins.gradle.org is no longer available for me.

It also sounds as if it is possible to publish to plugins.gradle.org from a Maven project, so a rewrite of this project to a Gradle build is not necessary to achieve that. Also, we have no need to maintain old/outdated Gradle versions, so I don't know why that was considered an immovable blocker. The reasons that the existing Gradle plugin needed to be abandoned don't seem to be sound.

The current model seems to be confusing people unnecessarily, so let's rethink this. I've read many threads now with people complaining about having this plugin split out. We also have a very large population of existing users using this plugin under the current coordinates, as @ben-manes mentioned.

I'd like to remove the deprecation warning and make whatever adjustments are necessary in the main repository. It would be great if you could help with that. We get great benefit having the Gradle plugin built as part of this project - it keeps all interfaces into jsonschema2pojo in sync, compatible, compiled and versioned together.

**Comment by eirnym** [comment]
Hi John,

It's actually published a day ago as there was a long review of this plugin.

As previously we discussed, my plugin become an official one, also I told, that this scheme is used in checker framework. Now you want to drop what we discussed just after I published a new version with a lot of fixes. Not transfer the project, not creating an organization for that (as proposed @hannesa2), just drop everything. Should I next just delete all plugins I published and delete the repo?

PS: 2020 was an intense year in a bad way, so I able to finish many things only in 2021.

**Comment by hannesa2** [comment]
> It would be great if you could help with that.

Please make @eirnym a maintainer/owner to avoid a single point of failure in the future

**Comment by hannesa2** [comment]
https://github.com/eirnym/js2p-gradle/issues/8#issuecomment-774412215

**Comment by joelittlejohn** [comment]
@hannesa2 That has been done, nearly a year ago.

@eirnym Sorry, I understand you are frustrated here. I can also empathise with having a busy/turbulent year and not having time to donate to open source for a while. Everyone's ability to donate that time will ebb and flow over the years. The past few years have been one crisis after another for me 😄 It's why I try never to demand the time of others or have others demand time from me. We have all given so much already (this project has been running for over 10 years, with a huge number of contributors in that time).

Clarity and consistency for end users is of the upmost importance to me, and IMO keeping this plugin as part of the main project is the right way to achieve that (I realise the checker framework has a different model, you've mentioned it here and elsewhere where people are confused). You initially gave a list of reasons that splitting this plugin out was required, but it has come to light that many of those were unsound. I should have been more engaged with verifying that at the time and this is my fault.

I realise there are costs to keeping things together too. It limits the ability of individual plugins to rapidly iterate and diverge (and that rapid iteration can be a good thing). However with a single centralised model we prioritise long-term consistency, continuity and a slower but longer kind of life.

I understand if you're fed up with the whole exercise now and you have no interest in continuing to contribute, but please do delete https://plugins.gradle.org/plugin/org.jsonschema2pojo if you can. I really want to reserve this identity for the core project. If you want to continue to develop under your own package name, I'm more than happy to continue to recommend your plugin as the preferred route if you want me to.

**Comment by eirnym** [comment]
@joelittlejohn 

So you have excuses for yourself, and don't accept same excuses from others. I'm frustrated with only that. Last year was a great problem for the whole world as COVID-19 and for me it was too. I never asked you for a help as well. This is not a paid work for me and I was never obliged to do any work in a short window of time despite a situation in my real life. This is an open source world, where I do my contributions on my own spare time if I have it.

**Comment by eirnym** [comment]
I gave a promise that I'd keep documentation and library version to a release version, and I keep it. Moreso, you haven't published any version of library, nor maven plugin during this time, so I see no problems for end users with keeping things separate. 

Additionally, I'd add a `version` parameter from the beginning, but jsonschema2pojo is not a tool.

**Comment by eirnym** [comment]
From all all projects I worked and methodologies I read (and there're lot of them), as you have a version control system, you can just delete old code with no obligations for others, just be explicit and publish an announcement. To  preserve a continuity, it's better to leave a documentation for a while how to use an old version and where to get a new one.

**Comment by eirnym** [comment]
And about your thoughts, I'd prefer you to share them and discuss. How do you prefer to discuss, publicly or privately, I'll live up to you.

**Comment by eirnym** [comment]
BTW, removal process from plugins.gradle.org is not a press of a button and process can take some time. Also some folks already use a new version and will have to change their projects just because of your last minute decision.

**Comment by joelittlejohn** [comment]
> @joelittlejohn
> 
> So you have excuses for yourself, and don't accept same excuses from others. I'm frustrated with only that. Last year was a great problem for the whole world as COVID-19 and for me it was too. I never asked you for a help as well. This is not a paid work for me and I was never obliged to do any work in a short window of time despite a situation in my real life. This is an open source world, where I do my contributions on my own spare time if I have it.

I was trying to say I understand your challenges here (and I face them myself too). I think we all have a right to focus on things outside open source. No excuse is needed.

Please start the process to delete. It's important we resolve this quickly to minimise disruption. This plugin has been available for only a few days. If you had shared your plan to take the org.jsonschema2pojo package, I would have cautioned against this.

We don't agree here, and maybe we never will. If you feel you can't continue to contribute (or you want to do your own thing) that's fine and I understand.

**Comment by AlexTrotsenko** [comment]
@eirnym, @joelittlejohn guys since I was involved in some earlier discussions related to deprication of the current maven-based  plugin I would like to add my 5 cents before newly published plugin is completely deleted from Gradle portal.

It's clear that there were some problematic issues as well as the reasons for them.

However, like @hannesa2, I also believe, that it's always better if there are several people making contribution to the project. 

But now the main question if you guys believe, that your further collaboration is possible:
- If you are ok to continue developing gradle plugin together - I believe it's always possible to find solution, which will fit everyone's needs. it might be repo in jsonschema2pojo organisation, were you both have admin rights or so.
- If you both think that you can't move on - then indeed it's better to completely separate two projects and avoid any confusion.

**Comment by hannesa2** [comment]
> However, like @hannesa2, I also believe, that it's always better if there are several people making contribution to the project.

More then this. To keep it alive for a long time. Out of our current horizon, eg > 5 years, it's better when at least 2 persons can access `settings` tab

>  it might be repo in jsonschema2pojo organisation, were you both have admin rights or so.

Total agree ! I want to see a community project, instead of a personal repository

**Comment by ben-manes** [comment]
Please do not try to pressure @joelittlejohn into giving ownership rights to others. He has maintained this project for over a decade, pre-Github back when Google Code was the platform of choice. He gives contributors access as collaborators who can commit directly, assist in PRs, and help triage issues. The few restrictions can be automated away, such as moving the release process to Github Actions.

I do not understand the hostility towards contributing to this project directly. Even if there is low activity, reaching out to the developers would have been met with enthusiastic support. Had I been ping'd, I would have helped marshal the plugin compatibility changes through. We were in no way trying to maintain Gradle 1.x compatibility - that is simply how old this project is! While forking is a wonderful benefit of OSS, an even better one is the ability to collaborate with others who are also charitably giving up their free time. Please try to work with @joelittlejohn rather than around him.

Deleting the plugin is not a long endeavor. It can be done within [7 days](https://plugins.gradle.org/docs/deleting) of publication within the portal, otherwise [requested](https://github.com/gradle/plugin-portal-requests/issues) as a github issue. @joelittlejohn's request was within that time frame.

I believe we can merge in the compatibility fixes and any other improvements through PRs and perform a release. The changes in this fork are minor so far, had we wanted legacy support that was possible, and no one is against incompatible improvements regardless. We can also discuss how to automate the release so that the plugin's build is in Gradle and on the portal. If this is simply about self branding by using one's own github account, then please honor the request to not the domain name in the plugin and you can still be redirected to as the preferred plugin author. Please try to work with the existing community.

**Comment by AlexTrotsenko** [comment]
@ben-manes  I believe your comment about pressuring @joelittlejohn into giving ownership rights to others was at least partially addressed to me. Thus I would like to clarify my point of view.

As for me @joelittlejohn has ultimate power to decide the future of original maven-based Gradle plugin. If I understood it correctly, unfortunately, he had no time & plans to maintain it further with the same quality as maven plugin, etc. E.g. it was never published to Gradle plugin portal, was not properly accessible from Gradle Kotlin DSL, etc. 

Thus community effort from @eirnym  has appeared to fix these issues. 

If @joelittlejohn  is now see how he can and want to maintain Gradle plugin by himself - then perhaps the situation has changed and there is no need in this discussion.
However if community contribution is an essential part of Gradle plugin sub-project - then I would suggest to keep the discussions open. It can have different form of collaboration e.g. @eirnym  can take lead in developing Gradle specific features and @joelittlejohn is taking care of domain-specic issue or whatever he wants/can to be responsible for. And for sure, in such case, @joelittlejohn  should have enough right to manage what he want to be responsible for, e.g. Gradle portal, etc.

So I believe it is clear that I do not have any intention of pushing anything,  but rather suggesting starting common point of further discussion.  

And yes, if @joelittlejohn prefers to take care of the Gradle plugin by himself it's also an option. As for me, the evolution of the project is what matters at the end.

**Comment by joelittlejohn** [comment]
I caught up with @eirnym last night. Thanks for all the comments all. I'd love it if people wanted to continue helping to update and improve the Gradle plugin in this project, but I also understand if people want to do their own thing and team up on something else. I find it strange to read about me wanting to do everything by myself, because this isn't the case at all. I don't see a single PR raised by the people on this thread (and I've always been keen to add _anyone_ that starts investing time as a contributor).

I think that this project gets great benefit from being a mono repo, and I'm also keen to continue development on the current plugin because it looks to me like it actually isn't as doomed as was originally implied. I'm convinced that keeping all this stuff together is the right way to go. Of course this will mean our first tasks are dropping support for Gradle < 5.6, publishing to plugins.gradle.org and getting rid of these Gradle 7 deprecation warnings.

I'm gutted that this wave of enthusiasm hasn't gone into the current plugin. I would be more than happy for anyone on this thread to take a lead on development there as you suggest Alex. However I also don't think it's right to arrive at an open source project and make your contribution provisional on demands to fundamentally change how the project is structured and organised, before you've really contributed anything at all. The current structure is optimised to guarantee that when people want to add their own feature, they are required to think of all users. The plugins are compiled together, versioned together, and benefit from all the automated housekeeping of the core project. This is a model that's optimised for very few people to be able to maintain something for a very long time. What you'll find is that very few people continue to contribute to a project after their initial needs are met. I accept that it isn't optimised for rapid iteration in one area.

Maybe we'll move to a GitHub org in future, maybe not. We have a mono repo so the benefits are minimal, and if I get hit by a bus tomorrow there are 8 people who can continue to modify this project and/or move it entirely.

I respect what everyone is building. So if anyone feels so inclined, feel free to dive in and start improving and updating the plugin in this project. However I think the best way forward is for you guys to revert to your original package at plugins.gradle.org and delete this: https://plugins.gradle.org/plugin/org.jsonschema2pojo. That identity is owned by the core.

I don't have any animosity towards anyone, despite being sad that things were split off in this way, so please don't hesitate to raise things in future.

**Comment by eirnym** [comment]
I put my 2c here from talk with @joelittlejohn as well. 

I don't see benefits of monorepo as plugin can have updates independently from core library and this will hold fixes to a plugin. 

Versioning is not an issue as can be seen in other opensource apps. 

Same org will clear any disambiguation where to look for plugin source.

**Comment by eirnym** [comment]
One more important note on collaborators vs members of an GitHub organisation. 

At first as I said, people won't be confused where to search for every part of the project. 

At second and more important is management flexibility. Collaborators in personal project can do nothing if the owner is absent for a while. So yes, they can do pull requests and resolve issues, that's it. No new releases and no way to manage if needed. 

And at last, but not least is about feature-parity. It's about management, not about "same repository"

**Comment by ben-manes** [comment]
> Collaborators in personal project can do nothing if the owner is absent for a while. So yes, they can do pull requests and resolve issues, that's it. No new releases and no way to manage if needed.

Thankfully that is not accurate. A collaborator may create Github releases, along with other actions. The publishing of artifacts depends on the setup and is unrelated to Github permissions. In many projects the creation of a release will automatically publish a new version. For the `gradle-versions-plugin`, I had a contributor set this up with TravisCI (who led the project for over a year) and I later migrated that to Github Actions. I often give the authors of PRs contributor permissions and asked them to handle the release themselves. Setting this up for this repository would be straightforward and @joelittlejohn might prefer it (as no need to mess with local setup issues again).

The main limitation of this role is the inability to add (a) secrets and (b) collaborators. That makes sense from a security perspective but does mean an absent owner has some drawbacks, which are easily resolved by a hard fork. The other limitations, like deleting the repository, are better left to the owner. This is all not applicable to @joelittlejohn who has been responsive to questions, even during inactive development. What are the reasons for why multiple owners are needed? This issue has been riddled with inaccurate statements being asserted as undeniable facts, both technical and non-technical, which I am finding very troubling and frustrating.

The benefits of different organization structures, automating releases, etc. can be tackled in new issues if you would like to discuss them specifically. Obviously @joelittlejohn will make a decision and I trust his judgement as he has always shown himself to be open-minded and conscious of his users' needs.

**Comment by hannesa2** [comment]
> What are the reasons for why multiple owners are needed?

A single owner is mostly the reason of a dead project

**Comment by KengoTODA** [comment]
Hi there 👋 
I'm a contributor for the Gradle plugin developed under the namespace github.com/eirnym. It seems that we have several points to discuss:

1. The Gradle plugin under github.com/eirnym is official or not
2. Monorepo or not
3. Organization or personal account

Till now I thought that github.com/eirnym is official. If so, we have no reason to remove the Gradle plugin published with the ID `org.jsonschema2pojo`. So I think we need to make this point clear first.  
I'm not so clear about the process behind of making the Gradle plugin under github.com/eirnym. I personally hope that good FOSS community is usually open and has welcoming culture for such valuable and pioneer contributions.

About the 2nd topic, I strongly suggest to use an isolated repository for Gradle plugin. I did so for [spotbugs-gradle-plugin](https://github.com/spotbugs/spotbugs-gradle-plugin/) because it needs to follow two lifecycles: Gradle and the target tool. It should have frequent release process and test methodology such as release branches for old Gradle, [build matrix for supported Gradle versions](https://github.com/spotbugs/spotbugs-gradle-plugin/blob/2af57527b23592fa90b16d7433ff3598a32178d1/.github/workflows/gradle.yml#L17), etc. At least [jsonschema2pojo's release process is not enough frequent as Gradle plugin](https://github.com/joelittlejohn/jsonschema2pojo/releases) I believe.

The 3rd question is something trivial to me, but if the plugin is official and we choose isolated repository, then it's nice to have an organization and make a repository for the Gradle plugin in it.

I agree with @hannesa2, multi-admins is better because I worked to [fork FindBugs project that had single inactive admin](https://mailman.cs.umd.edu/pipermail/findbugs-discuss/2017-September/004383.html). But at this moment it's OK to have only one organization admin. Just make sure that the repository for Gradle plugin has multi-admins, then problem in this context will be solved.  
The whole community still has the same problem, but it's better to make another place to discuss for it. As mentioned by @ben-manes, we pay respect to @joelittlejohn's work for long years, and such a big judgement should be made with his will and understanding.


It is my opinion at this moment. Now [removal process is running](https://github.com/gradle/plugin-portal-requests/issues/85) but I hope we can build a consensus about how community will collaborate from now on.

**Comment by ben-manes** [comment]
Hi @KengoTODA,

This issue has become toxic and rather than rehash the frustrations, can we discuss steps to move forward and rebuild trust? I propose the following.

### Can we merge the improvements into this project's plugin?
This would resolve @joelittlejohn concern of contributors disappearing and the inability to have new features sync'd across plugins. The forked plugin has very few actual changes from the one here. It was not a rewrite.

### Can we automate the release process so that the plugin uses a Gradle build, rather than integrate with the Maven one?
This would avoid quirkiness of trying to build with Maven, allow for releasing to the plugin portal, and be more natural for further plugin development.

### Do we want to allow the releases to diverge?
This would allow the core and plugins do not wait for each other as they each make improvements at their own pace.

### At some future date: If contributors prove themselves, can we promote to a Github organization?
If trust is built by active, long standing collaboration then this redundancy is beneficial. It helps avoid the bus factor concerns and if @joelittlejohn ever does decide to leave, he knows the community is being taken care of.

### At some future date: If desired, can we split the repository apart?
If a mono-repo is not beneficial then restructuring might better fit the team. However, this requires a team of active contributors. Currently we should structure for @joelittlejohn's benefit as he is the only one who has made a multi-year commitment.

**Comment by hannesa2** [comment]
Nothing is toxic, when people write there opinions and wishes.

I guess _everyone_ has to move, otherwise it will end in a spilt forever probably or a dead Gradle project. All of them are sad.

At least for me it's clear, the time of absolutism is over and we live in the present, not in the future. 
"At some future date" will not happen, do it now

**Comment by joelittlejohn** [comment]
@KengoTODA Thanks for your comment. As another plugin maintainer I do value your opinion so please don't think your comment has been ignored. I'm keen to stick with the current project setup for now, but I never say 'never'.

Ironically, as part of bringing a variety of plugins in the main build up to date today I have actually had to bring the Gradle plugin substantially up to date already (hence closing this issue).
